bin/main

```diff
@@ -1,163 +1,110 @@
-@player2 = gets.chomp
-
-def display_board
-  puts @board[0] + '|' + @board[1] + '|' + @board[2]
-  puts @board[3] + '|' + @board[4] + '|' + @board[5]
-  puts @board[6] + '|' + @board[7] + '|' + @board[8]
-end
-
-def players_turn
-  puts 'choose from 1 - 9'
-  turn = gets.chomp.to_i
-  # if turn is not in the range of 1-9 return invalid
-
-  arr = (1..9)
-  until arr.include? turn
-    display_board
-    puts 'invalid choose from 1 - 9'
-    turn = gets.chomp.to_i
+class GameLogic
+  def initialize(board, current_player)
+    @board = board
+    @current_player = current_player
   end
-  turn -= 1
-  while @board[turn] != '_'
-    display_board
-    puts 'Slot already taken or invalid, choose from 1 - 9'
-    turn = gets.chomp.to_i
-    turn -= 1
+
+  def validate_wins(pos_a, pos_b, pos_c, player2, player1)
+    if @board[pos_a] == @board[pos_b] && @board[pos_b] == @board[pos_c] && @board[pos_a] != '_'
+      if @current_player == 'O'
+        puts "#{player2} wins"
+      else
+        puts "#{player1} wins"
+      end
+      true
+    end
   end
-  @board[turn] = @current_player
 end
 
-def flip_player
-  if @current_player == 'X'
-    @current_player = 'O'
-    puts "It's #{@player2}'s turn to play"
-  else
+class TicTacToe
+  def initialize
+    @board = %w[
+      _ _ _
+      _ _ _
+      _ _ _
+    ]
+    @game_on_going = true
     @current_player = 'X'
-    puts "It's #{@player1}'s turn to play"
+    puts 'Player1 enter your name'
+    @player1 = gets.chomp
+    puts 'Player2 enter your name'
+    @player2 = gets.chomp
   end
-end
-
-def check_win
-  check_row
-  check_column
-  check_diagonal
-end
 
-def check_column
-  if @board[0] == @board[3] && @board[0] == @board[6] && @board[0] != '_'
-    @game_on_going = false
-    if @current_player == 'O'
-      puts "#{@player2} wins"
-    else
-      puts "#{@player1} wins" 
-    end
-  elsif @board[1] == @board[4] && @board[1] == @board[7] && @board[1] != '_'
-      @game_on_going = false
-      if @current_player == 'O'
-        puts "#{@player2} wins"
-      else
-        puts "#{@player1} wins" 
-      end
-  elsif @board[2] == @board[5] && @board[2] == @board[8] && @board[2] != '_'
-      @game_on_going = false
-      if @current_player == 'O'
-        puts "#{@player2} wins"
-      else
-        puts "#{@player1} wins" 
-      end
+  def display_board
+    puts @board[0] + '|' + @board[1] + '|' + @board[2]
+    puts @board[3] + '|' + @board[4] + '|' + @board[5]
+    puts @board[6] + '|' + @board[7] + '|' + @board[8]
   end
-end
 
-def check_row
-  if @board[0] == @board[1] && @board[0] == @board[2] && @board[0] != '_'
-    @game_on_going = false
-    if @current_player == 'O'
-      puts "#{@player2} wins"
-    else
-      puts "#{@player1} wins" 
+  def players_turn
+    puts 'choose from 1 - 9'
+    turn = gets.chomp.to_i
+    arr = (1..9)
+    until arr.include? turn
+      display_board
+      puts 'invalid choose from 1 - 9'
+      turn = gets.chomp.to_i
     end
```

Comment:
You will have to modify this method, too. I suggest you leave the `until arr.include?` part and call it in main.rb with the player's choice as an argument.